src/index.js

```diff
@@ -1,59 +1,96 @@
-import qsm from 'query-string-manipulator'
+const EXPANSION_REGEX = /(?<=\{)[^{]*(?=\})/g
 
-const HAL_EXPANSION_TOKEN = '{?'
-const HAL_CONTINUATION_TOKEN = '{&'
-const tokens = [HAL_EXPANSION_TOKEN, HAL_CONTINUATION_TOKEN]
+const DEFAULT_OPERATOR =
+  { operator: '', separator: ',', named: false, ifEmpty: false }
+const OPERATORS = [
+  { operator: '#', separator: ',', named: false, ifEmpty: false },
+  { operator: '.', separator: '.', named: false, ifEmpty: false },
+  { operator: '/', separator: '/', named: false, ifEmpty: false },
+  { operator: ';', separator: ';', named: true, ifEmpty: false },
+  { operator: '?', separator: '&', named: true, ifEmpty: true },
+  { operator: '&', separator: '&', named: true, ifEmpty: true }
+]
 
-function getOptionsToken (link) {
-  for (const token of tokens) {
-    const index = link.indexOf(token)
-    if (index !== -1) {
-      return [token, index]
-    }
+function getNamedOperatorIfNeeded (operator, varName, el) {
+  if (operator.named) {
+    return `${varName}=${el}`
+  }
+  return el
+}
+
+function getValueFor (args, varName, operator) {
+  if (args[varName] instanceof Array) {
+    return args[varName]
+      .flatMap(el => getNamedOperatorIfNeeded(operator, varName, el))
+      .join(operator.separator)
+  }
+  return getNamedOperatorIfNeeded(operator, varName, [`${args[varName]}`]
+    .join(operator.separator))
+}
+
+function containsArgument (args, v, canBeEmpty) {
+  if (canBeEmpty) {
+    return !!args && typeof args[v] !== 'undefined'
   }
-  return [null, 0]
+  return !!args && !!args[v]
 }
 
 function getLinkOptions (link) {
-  const [token, tokenIndex] = getOptionsToken(link)
-  if (token) {
-    const strippedLink = link.substr(0, tokenIndex)
-    const optionsStr = link.substr(tokenIndex + token.length).split('}')[0]
+  const groups = link.match(EXPANSION_REGEX)
+  if (groups) {
     return {
-      url: strippedLink,
-      options: optionsStr.split(',')
+      url: link,
+      options: groups.flatMap(str => str.split(','))
```

Comment:
Hey, I'm not sure about adopting [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) and [startsWith](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith). They require polyfill to support IE. Whats your opinion on this, @SBylemans ?

Reply:
That depends, do you still wanna support IE? 
In my opinion, you don't need to do that. You can put a disclaimer or something that when using this lib for IE they need to polyfill themselves. Or we could add a polyfill just in case.

Reply:
It's not just IE. `flatMap` was introduced to Mrdkosoft Edge this January. Chrome & Firefox September 2018. I don't believe that there was enough time to upgrade every browser. Hey don't worry about this. I'll refactor it just to be on the safe side.

Reply:
Thx! Any timeline on when you would be able to release then?